infrastructure/terraform/components/app/amplify_branch_deploy.tf

```diff
@@ -0,0 +1,12 @@
+resource "null_resource" "amplify_branch_deploy" {
+  # Using triggers to force execution on every apply
+  triggers = {
+    always_run = timestamp()
+  }
+
+  depends_on = [module.amplify_branch]
+
+  provisioner "local-exec" {
+    command = "aws amplify start-job --app-id ${aws_amplify_app.main.id} --branch-name ${var.branch_name} --commit-message \"${replace(var.deployment_description, "\"", "\\\"")}\" --commit-id ${var.commit_id} --job-type RELEASE"
```

Comment:
There are two 'branch' module instantiations - one on the 'app' component and one on the 'branch' component. The latter used for the PR branch deployments, or any other non-main branch you want to deploy, and still has auto-build enabled.

The idea is that the main branch is deployed to the app component 'production' branch, but that deployment only triggers after the code has passed the pipeline tests, etc, rather than as soon as the commit is pushed which is the current behaviour.

This also allows ANY commit on the main branch to be deployed (i.e. something with a tag on the main branch) which gives us the ability to roll back too. You can kind of do this in the Amplify console if you have a build from that commit, but those won't be associated with tags/releases in the same way.